core/src/main/scala/scalafix/rewrite/Xor2Either.scala

```diff
@@ -13,16 +14,90 @@ case object Xor2Either extends Rewrite {
     }
     object DType extends Desugared[Type]
     object DTerm extends Desugared[Term]
+
+    //If we start making many of these, would it be worth having case classes with some structure?
+    //Possibly if I could select the appropriate based on a match, I have not been able to do that yet.
+    case class ReplaceType(original: Type,
+                           replacement: Type,
+                           newString: String)
+    val xorT: ReplaceType =
+      ReplaceType(t"cats.data.XorT", t"scala.util.EitherT", "EitherT")
+    val xor: ReplaceType =
+      ReplaceType(t"cats.data.Xor", t"scala.util.Either", "Either")
+    val xorLeft: ReplaceType =
+      ReplaceType(t"cats.data.Xor.Left", t"scala.util.Left", "Left")
+    val xorRight: ReplaceType =
+      ReplaceType(t"cats.data.Xor.Right", t"scala.util.Either.Right", "Right")
+
+    //experimenting to see if I can create lists of types to try to match.  No luck so far.
+    val replacementTypes: List[ReplaceType] =
+      List(xorT, xor, xorLeft, xorRight)
+
+    val termMap: Map[String, String] = Map(
+      "cats.data.Xor.Right.apply" -> "Right",
+      "cats.data.Xor.Left.apply" -> "Left"
+    )
+
     // NOTE. This approach is super inefficient, since we run semantic.desugar on
     // every case for every node in the tree. Ideally, we first match on the
     // syntax structure we want and then run semantic.desugar.
-    ast.collect {
-      case t @ DType(t"cats.data.Xor") =>
-        Patch(t.tokens.head, t.tokens.last, s"Either")
+    val tpes: Seq[(Type, ReplaceType)] = ast.collect {
+      case t @ DType(t"cats.data.XorT") => t -> xorT
+      case t @ DType(t"cats.data.Xor") => t -> xor
+      case t @ DType(t"cats.data.Xor.Left") => t -> xorLeft
+      case t @ DType(t"cats.data.Xor.Right") => t -> xorRight
+    }
+
+    //This is unsafe, come up with something better
+    val typeReplacements = tpes.map {
+      case (tree, rt) => Patch.replace(tree, rt.newString)
+    }
+
+    val terms: Map[String, Term] = ListMap(ast.collect {
       case t @ DTerm(q"cats.data.Xor.Right.apply[..$_]") =>
-        Patch(t.tokens.head, t.tokens.last, s"Right")
+        "cats.data.Xor.Right.apply" -> t
       case t @ DTerm(q"cats.data.Xor.Left.apply[..$_]") =>
-        Patch(t.tokens.head, t.tokens.last, s"Left")
+        "cats.data.Xor.Left.apply" -> t
+    }: _*)
+
+    val termReplacements = terms.map {
+      case (key, term) => Patch.replace(term, termMap(key))
+    }
+
+    val allImports = ast.collect {
+      case t @ q"import ..$importersnel" => t -> importersnel
     }
+
+    val firstImport = allImports.head._1
+    val firstImportFirstToken = firstImport.tokens.head
+    val tokenBeforeFirstImport =
+      ast.tokens.takeWhile(_ != firstImportFirstToken).lastOption
```

Comment:
I also need to be smarter about existing imports in general.  Since I am collecting up all imports and patches, I could just find imports needed per patch that are not already imported.  Maybe I need to switch to a structure with more information than just patch to make that convenient.